deps/ncrypto/ncrypto.cc

```diff
@@ -1852,6 +1857,108 @@ DataPointer pbkdf2(const Digest& md,
   return {};
 }
 
+#if OPENSSL_VERSION_MAJOR >= 3 && OPENSSL_VERSION_MINOR >= 2
+#ifndef OPENSSL_NO_ARGON2
+namespace {
+
+class MaxThreadsScope final {
+ public:
+  MaxThreadsScope(uint64_t threads) : ctx_{OSSL_LIB_CTX_new()} {
+    OSSL_set_max_threads(ctx_.get(), threads) == 1;
+  }
+  ~MaxThreadsScope() { OSSL_set_max_threads(ctx_.get(), 0); }
```

Comment:
Should there be some protection against having multiple `MaxThreadScope` instances on the stack? At the very least, this should have some comments warning against it.

Reply:
FWIW when I change this to actually assert the result

```
NCRYPTO_ASSERT_TRUE(OSSL_set_max_threads(ctx_.get(), threads) == 1)
```

Then the tests start failing. So I'm not sure everything wrt. threads here is OK.